### PR TITLE
[CLOB-177] add e2e tests for rejecting long-term IOC/FOK in CheckTx and DeliverTx

### PR DESCRIPTION
**Pull request by jakob-dydx**

### CODE REVIEW

**Review by jayy04 — APPROVED** [review]

File: protocol/x/clob/e2e/long_term_orders_test.go [hunk 1]
```diff
@@ -187,3 +187,49 @@ func TestCancelStatefulOrder(t *testing.T) {
...
+	// Reject long-term IOC/FOK in DeliverTx
+	blockAdvancement := testapp.BlockAdvancementWithErrors{
+		BlockAdvancement: testapp.BlockAdvancement{
+			StatefulOrders: []clobtypes.Order{
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_IOC,
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_FOK,
+			},
+		},
+		ExpectedDeliverTxErrors: map[int]string{
+			0: clobtypes.ErrStatefulOrdersCannotRequireImmediateExecution.Error(),
```
Comment (lucas-dydx) [inline protocol/x/clob/e2e/long_term_orders_test.go:230]: Can we rename [this error](https://github.com/dydxprotocol/v4-chain/blob/8f43054603ed7595b49cef40ba8dd38aa708c3de/protocol/x/clob/types/errors.go#L350) to `ErrLongTermOrdersCannotRequireImmediateExecution`?

File: protocol/testutil/app/block_advancement.go [hunk 2]
```diff
@@ -49,10 +49,10 @@ func (b BlockAdvancementWithError) AdvanceToBlock(
 		) (haltchain bool) {
 			expectedError, found := b.ExpectedDeliverTxErrors[txIndex]
 			if found && expectedError != "" {
-				require.True(t, response.IsErr())
+				require.Conditionf(t, response.IsErr, "Expected CheckTx to error. Response: %+v", response)
```
Comment (lucas-dydx) [inline protocol/testutil/app/block_advancement.go:52]: For my understanding why is `Conditionf` preferred over using `True` + calling the function?

**Review by lucas-dydx — APPROVED** [review]

File: protocol/x/clob/e2e/long_term_orders_test.go (hunk 1 above)
Comment (jakob-dydx) [reply protocol/x/clob/e2e/long_term_orders_test.go:230]: yep will do

File: protocol/testutil/app/block_advancement.go (hunk 2 above)
Comment (jakob-dydx) [reply protocol/testutil/app/block_advancement.go:52]: I'll just change to use `require.True` again

**Pull request merged**